catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -39,92 +48,105 @@
 import ddf.catalog.validation.ValidationException;
 
 public class MetacardValidityMarkerPlugin implements PreIngestPlugin {
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(MetacardValidityMarkerPlugin.class);
+
+    private static final Logger INGEST_LOGGER =
+            LoggerFactory.getLogger(Constants.INGEST_LOGGER_NAME);
+
+    private final Predicate<Object> didNotFailEnforcedValidator = Objects::nonNull;
 
     private List<String> enforcedMetacardValidators;
 
     private List<MetacardValidator> metacardValidators;
 
-    private static final Logger LOGGER =
-            LoggerFactory.getLogger(MetacardValidityMarkerPlugin.class);
+    @Override
+    public CreateRequest process(CreateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Metacard> validatedMetacards = validateList(input.getMetacards(), Function.identity());
+        return new CreateRequestImpl(validatedMetacards,
+                input.getProperties(),
+                input.getStoreIds());
+    }
 
-    public static final String VALIDATION_ERRORS = BasicTypes.VALIDATION_ERRORS;
+    @Override
+    public UpdateRequest process(UpdateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Map.Entry<Serializable, Metacard>> validatedUpdates = validateList(input.getUpdates(),
+                Map.Entry::getValue);
+        return new UpdateRequestImpl(validatedUpdates,
+                input.getAttributeName(),
+                input.getProperties(),
+                input.getStoreIds());
+    }
 
-    public static final String VALIDATION_WARNINGS = BasicTypes.VALIDATION_WARNINGS;
+    private <T> List<T> validateList(List<T> requestItems, Function<T, Metacard> itemToMetacard) {
+        Map<String, Integer> counter = new HashMap<>();
```

Comment:
Two thoughts on the `counter` map:
1. If you were ever to want to parallelize the stream processing you'd need to make this a concurrent collection
2. Instead of the `merge` operations you're performing on the map values, you might consider just changing out to a `Map<String, AtomicInteger>` and calling one of its increment methods. I think that would read a bit cleaner.